### PR TITLE
feat(surveys): API (headless) type mode

### PR DESCRIPTION
**Pull request by liyiy**
## Problem

We want users to be able to create surveys still through our app but allow them to customize their own survey components. 

To go in with https://github.com/PostHog/posthog-js/pull/736

## Changes

Add "API" mode where there's no appearance edit options. Users can fill out the "question" part if they choose to so they have a reference to it, but it's not required to create an "API" mode type survey

Screen Shot 2023-07-11 at 3 29 15 PM

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/surveys/surveyLogic.tsx [hunk 1]
```diff
@@ -269,10 +269,10 @@ export const surveyLogic = kea<surveyLogicType>([
     forms(({ actions, props, values }) => ({
         survey: {
             defaults: { ...NEW_SURVEY } as NewSurvey | Survey,
-            errors: ({ name, questions }) => ({
+            errors: ({ name, questions, type }) => ({
                 name: !name && 'Please enter a name.',
                 questions: questions.map((question) => ({
-                    question: !question.question && 'Please enter a question.',
+                    question: type !== SurveyType.API && !question.question && 'Please enter a question.',
```
Comment (neilkakkar) [inline frontend/src/scenes/surveys/surveyLogic.tsx:275]: I think we have different ideas for what the headless mode does - let's chat during stand up!

**Comment by liyiy** [comment]
Added a api response preview to it too with exposed survey ID.

Screen Shot 2023-07-12 at 1 40 45 PM

**Comment by neilkakkar** [comment]
Recommendation: The API response ought to be what they'll get with `getActiveMatchingSurveys`, which has fewer fields than the projects/<id>/survey api response. Just so it's not misleading what other fields they can filter on

**Comment by liyiy** [comment]
> Recommendation: The API response ought to be what they'll get with `getActiveMatchingSurveys`, which has fewer fields than the projects//survey api response. Just so it's not misleading what other fields they can filter on

oops yes! will fix

**Pull request merged**